ext/RMagick/rmimage.c

```diff
@@ -3854,27 +4190,45 @@ Image_constitute(VALUE class ATTRIBUTE_UNUSED, VALUE width_arg, VALUE height_arg
         rb_raise(rb_eNoMemError, "not enough memory to continue.");
     }
 
+#if defined(IMAGEMAGICK_7)
+    exception = AcquireExceptionInfo();
+    SetImageExtent(new_image, width, height, exception);
+#else
     SetImageExtent(new_image, width, height);
     exception = &new_image->exception;
```

Comment:
These changes where made to prevent leaking of pixels.v. But it does look like you found a mistake here. Very good catch. For IM6 the exception will be freed and will result in an access violation. Will push a patch to fix this.